### PR TITLE
Add auth wrapper components & Fetch model UUID on details page

### PR DESCRIPTION
**Pull request by hatched**
## Done

- When trying to view an authenticated page, if you've not already authenticated it'll now render a very basic text link for you to authenticate with usso.
- Get the model UUID and pass it to the Terminal component for use later.

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Open / it should request that you log in.
- After logging in, click one of the models in your model list to show the details page.
- The page with dummy data should load as expected.

### CODE REVIEW

File: src/app/selectors.js [hunk 1]
```diff
@@ -36,3 +38,32 @@ export const getDecodedMacaroons = macaroons => {
...
+
+/**
+  Gets the model UUID from the supplied name.
+  @param {String} name The name of the model.
+  @param {Object} modelInfo The list of models.
+  @returns {Object|Null} The model UUID or null if none found.
+*/
+export const getModelUUIDByName = (name, modelInfo) => {
+  if (modelInfo) {
+    for (let uuid in modelInfo) {
```
Comment (squidsoup) [inline src/app/selectors.js]: `filter` might be nicer.

File: src/components/IsLoggedIn/IsLoggedIn.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return root.visitURL;
+      }
+    }
+  });
+
+  if (!userIsLoggedIn) {
+    return (
+      <div>
+        Please{" "}
+        <a href={visitURL} rel="noopener noreferrer" target="_blank">
```
Comment (squidsoup) [inline src/components/IsLoggedIn/IsLoggedIn.js]: This should be a vanilla link component.

File: src/app/selectors.js (hunk 1 above)
Comment (hatched) [reply src/app/selectors.js]: I want to bail out as soon as it's found so `some` would be preferred, however `modelInfo` is an Object so can't filter over it. In order to use any of the array utilities I'd have to first do something like `Object.keys().some(key => { ... })` which wouldn't be as performant in the event that there is a significant number of models.

File: src/components/IsLoggedIn/IsLoggedIn.js (hunk 2 above)
Comment (hatched) [reply src/components/IsLoggedIn/IsLoggedIn.js]: This is only a temporary page to make qa'ing and developing easier. There is a design for this page that's not this :D

**Pull request merged**